phoebe/solverbackends/solverbackends.py

```diff
@@ -2247,6 +2261,240 @@ def valid_options(self):
         return ['maxiter', 'gtol', 'norm']
 
 
+class Differential_CorrectionsBackend(BaseSolverBackend):
+    """
+    See <phoebe.parameters.solver.optimizer.differential_corrections>.
+
+    The run method in this class will almost always be called through the bundle, using
+    * <phoebe.frontend.bundle.Bundle.add_solver>
+    * <phoebe.frontend.bundle.Bundle.run_solver>
+    """
+    def __init__(self):
+        super().__init__()
+        self._allow_mpi = False
+
+    @property
+    def workers_need_solution_ps(self):
+        # TODO: only if progress_every_niters?
+        return False
+
+    def run_checks(self, b, solver, compute, **kwargs):
+        solver_ps = b.get_solver(solver=solver, **_skip_filter_checks)
+        if not len(solver_ps.get_value(qualifier='fit_parameters', fit_parameters=kwargs.get('fit_parameters', None), expand=True)):
+            raise ValueError("cannot run differential_corrections without any parameters in fit_parameters")
+
+    def _get_packet_and_solution(self, b, solver, **kwargs):
+        # NOTE: b, solver, compute, backend will be added by get_packet_and_solution
+        solution_params = []
```

Comment:
Could `solution_params` with all common parameters across all solvers be moved to the base class (`BaseSolverBackend`)?

Reply:
Yes, I have a ticket to try to refactor some of this logic.  We have a base class for scipy optimizers, but it will take more work to split out the base logic that applies to other optimizers as well.